go/uidmap/uidmap.go

```diff
@@ -285,27 +292,33 @@ func (u *UIDMap) MapUIDsToUsernamePackages(ctx context.Context, g libkb.UIDMappe
 	apiLookupIndex := make(map[int]int)
 
 	var uidsToLookup []keybase1.UID
-
-	for i, uid := range uids {
-		up, status := u.findUsernamePackageLocally(ctx, g, uid, fullNameFreshness, forceNetworkForFullNames)
-
-		// If we successfully looked up some of the user, set the return slot here.
-		if up != nil {
-			res[i] = *up
-		}
-
-		// There are 3 important cases when we should go to network:
-		//
-		//  1. No username is found (up == nil)
-		//  2. No FullName found and we've asked to force network lookups (status == notFound && forceNetworkForNullNames)
-		//  3. The FullName found was stale (status == stale).
-		//
-		// Thus, if you provide forceNetworkForFullName=false, and fullNameFreshness=0, you can avoid
-		// the network trip as long as all of your username lookups hit the cache or are hardcoded.
-		if up == nil || (status == notFound && forceNetworkForFullNames) || (status == stale) {
+	if u.testNoCachingMode {
+		for i, uid := range uids {
 			apiLookupIndex[len(uidsToLookup)] = i
 			uidsToLookup = append(uidsToLookup, uid)
 		}
+	} else {
+		for i, uid := range uids {
+			up, status := u.findUsernamePackageLocally(ctx, g, uid, fullNameFreshness, forceNetworkForFullNames)
+			g.GetLog().CDebugf(ctx, "| findUsernamePackageLocally(%s): %v", uid, status)
+			// If we successfully looked up some of the user, set the return slot here.
+			if up != nil {
+				res[i] = *up
+			}
+
+			// There are 3 important cases when we should go to network:
+			//
+			//  1. No username is found (up == nil)
+			//  2. No FullName found and we've asked to force network lookups (status == notFound && forceNetworkForNullNames)
+			//  3. The FullName found was stale (status == stale).
+			//
+			// Thus, if you provide forceNetworkForFullName=false, and fullNameFreshness=0, you can avoid
+			// the network trip as long as all of your username lookups hit the cache or are hardcoded.
```

Comment:
`|| testNoCachingMode` here?

Reply:
Keep in mind I was also racing again multiple invokations of `service.TeamGet`, where I needed to make sure to increment the time after every try. That wasn't looking so easy, so I decided to take the blunt approach with the caches here.

Reply:
`pollForMembershipUpdate` this guy just spam runs `service.TeamGet`

Reply:
i think my approach is just as blunt?

Reply:
Well you also need to change line 305? Maybe less clean? Your call.

Reply:
no, it'll get trampled over

Reply:
```
	// Always write these results out if the cached value is unset.
				// Or, see below for other case...
				writeResults := res[apiLookupIndex[i]].NormalizedUsername.IsNil()
```

Reply:
```
if writeResults {
					// Overwrite the row with whatever was returned from the server.
					res[apiLookupIndex[i]] = row
				}
```